lib/chef/knife/yaml_convert.rb

```diff
@@ -81,7 +81,13 @@ def resource_hash_to_string(resource_hash, filename)
 
       ruby_contents << "#{type} \"#{name}\" do"
       r.each do |k, v|
-        ruby_contents << "  #{k} \"#{v}\""
+        ruby_contents <<
+          case v
+          when Array
+            "  #{k} #{v}"
+          else
+            "  #{k} \"#{v}\""
+          end
```

Comment:
I think what you want here is probably just `"  #{k} #{v.inspect}"`

Reply:
(instead of the whole case statement)

Reply:
Good call, that is better.